workers/generic-worker/process/multiuser_darwin.go

```diff
@@ -0,0 +1,228 @@
+//go:build multiuser
+
+package process
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+type (
+	Command struct {
+		// ResourceMonitor is a function that monitors the system's resource usage.
+		// It should send the resource usage data to the first channel of type
+		// *ResourceUsage and stop measuring usage when the second channel of
+		// type struct{} is closed.
+		ResourceMonitor func(chan *ResourceUsage, chan struct{})
+		mutex           sync.RWMutex
+		*exec.Cmd
+		// abort channel is closed when Kill() is called so that Execute() can
+		// return even if cmd.Wait() is blocked. This is useful since cmd.Wait()
+		// sometimes does not return promptly.
+		abort chan struct{}
+		// Once command has run, Result is updated (similar to cmd.ProcessState)
+		result *Result
+		conn   net.Conn
+	}
+
+	CommandRequest struct {
+		Path    string   `json:"path"`
+		Args    []string `json:"args"`
+		Env     []string `json:"env"`
+		Dir     string   `json:"dir"`
+		Stdin   bool     `json:"stdin"`
+		Stdout  bool     `json:"stdout"`
+		Stderr  bool     `json:"stderr"`
+		Setctty bool     `json:"setctty"`
+		Setpgid bool     `json:"setpgid"`
+		Setsid  bool     `json:"setsid"`
+	}
+)
+
+func connectWithRetry(socketPath string, timeout time.Duration) (net.Conn, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("timeout reached waiting for launch agent socket: %w", ctx.Err())
+		case <-ticker.C:
+			if _, err := os.Stat(socketPath); errors.Is(err, os.ErrNotExist) {
+				// Socket file not yet present — skip connection attempt
+				continue
+			}
+			conn, err := net.Dial("unix", socketPath)
+			if err == nil {
+				return conn, nil // success!
+			}
+			log.Print(err)
+		}
+	}
+}
+
+func WriteFrame(w io.Writer, payload []byte) error {
+	var lenBuf [4]byte
+	binary.BigEndian.PutUint32(lenBuf[:], uint32(len(payload)))
+	if _, err := w.Write(lenBuf[:]); err != nil {
+		return err
+	}
+	_, err := w.Write(payload)
+	return err
+}
+
+func ReadFrame(r io.Reader) ([]byte, error) {
+	var lenBuf [4]byte
+	if _, err := io.ReadFull(r, lenBuf[:]); err != nil {
+		return nil, err
+	}
+	length := binary.BigEndian.Uint32(lenBuf[:])
+	buf := make([]byte, length)
+	if _, err := io.ReadFull(r, buf); err != nil {
+		return nil, err
+	}
+	return buf, nil
+}
+
+func (c *Command) Start() error {
+
+	// If command is meant to run as current user, don't send it to the launch agent...
+	if c.SysProcAttr == nil || c.SysProcAttr.Credential == nil || c.SysProcAttr.Credential.Uid == 0 {
+		return c.Cmd.Start()
+	}
```

Comment:
What about a check for headless here as well? (and other places in this PR you do this?)